javaparser-core/src/main/java/com/github/javaparser/ast/validator/Java1_0Validator.java

```diff
@@ -124,6 +124,13 @@
             (n, reporter) -> reporter.report(n, "Switch expressions are not supported.")
     );
 
+
+    final Validator noInstanceOfPattern = new SimpleValidator<>(InstanceOfExpr.class,
+            n -> n.getPattern().isPresent(),
+            (n, reporter) -> reporter.report(n, "Use of patterns with instanceof is not supported.")
```

Comment:
One thing that could be improved for validations is to use a single treewalk that does all validations, instead of a walk for every validation. I wouldn't prioritize that very highly though.

Reply:
#2658